### PR TITLE
Add Middleware options for RetryHandler and RedirectHandler

### PR DESCRIPTION
**Pull request by dadheech115**
This PR contains work for - 

1. MiddelwareOptions protocol
2. Retry Handler options
3. Redirect Handler options
4. Unit tests

### CODE REVIEW

File: MSGraphClientSDK/MSGraphClientSDK/Middleware/Options/MSRedirectHandlerOptions.m [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        if(maxRedirects>UPPER_LIMIT_FOR_REDIRECTS)
+        {
+            NSDictionary *userInfo = @{
+                                       NSLocalizedDescriptionKey: MSErrorOperationUnsuccessfulString,
+                                       NSLocalizedFailureReasonErrorKey:[NSString stringWithFormat:MSErrorMaxRedirectsLimitExceededString,UPPER_LIMIT_FOR_REDIRECTS]
+                                       };
+            *error = [NSError errorWithDomain:MSErrorDomain code:-1 userInfo:userInfo];
+            return nil;
+        }
+        _maxRedirects = maxRedirects;
```
Comment (deepak2016) [inline MSGraphClientSDK/MSGraphClientSDK/Middleware/Options/MSRedirectHandlerOptions.m:32]: add middlewaerOptionsType

File: MSGraphClientSDK/MSGraphClientSDK/Middleware/Options/MSRetryHandlerOptions.m [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+@implementation MSRetryHandlerOptions
+
+@synthesize middlewareOptionsType;
+
+- (instancetype)init
+{
+    self = [super init];
+    if(self)
+    {
+        _delay = 3;
```
Comment (deepak2016) [inline MSGraphClientSDK/MSGraphClientSDK/Middleware/Options/MSRetryHandlerOptions.m]: call initWithDelay from init

File: MSGraphClientSDK/MSGraphClientSDK/Middleware/Implementations/RedirectHandler/MSRedirectHandler.m [hunk 3]
```diff
@@ -14,20 +15,29 @@ -(void)setRequest:(NSMutableURLRequest *)request;
...
 @property (nonatomic, strong) id<MSGraphMiddleware> nextMiddleware;
+@property (nonatomic, strong) MSRedirectHandlerOptions *redirectHandlerOptions;
 @end
 
 @implementation MSRedirectHandler
+
+- (instancetype)init
 {
-    NSInteger maximumRedirectAllowed;
+    self = [super init];
```
Comment (deepak2016) [inline MSGraphClientSDK/MSGraphClientSDK/Middleware/Implementations/RedirectHandler/MSRedirectHandler.m]: use initWithOptions method to initialize

File: MSGraphClientSDK/MSGraphClientSDK/Middleware/Implementations/RetryHandler/MSRetryHandler.m [hunk 4]
```diff
@@ -22,20 +20,28 @@ -(void)setRequest:(NSMutableURLRequest *)request;
...
 
 @end
 
 @implementation MSRetryHandler
+
+- (instancetype)init
 {
-    NSInteger maxRetries;
+    self = [super init];
+    if(self)
```
Comment (deepak2016) [inline MSGraphClientSDK/MSGraphClientSDK/Middleware/Implementations/RetryHandler/MSRetryHandler.m]: use initWithOption method here as well.

**Review by deepak2016 — APPROVED** [review]

**Review by NakulSabharwal — APPROVED** [review]

**Pull request merged**